tests/system/test_query.py

```diff
@@ -161,6 +161,56 @@ class SomeKind(ndb.Model):
     assert [entity.foo for entity in results] == [-1, 0, 1, 2, 3, 4]
 
 
+def test_ancestor_query_with_namespace(client_context, dispose_of, other_namespace):
+    class Dummy(ndb.Model):
+        foo = ndb.StringProperty(default="")
+
+    entity1 = Dummy(foo="bar", namespace="xyz")
+    parent_key = entity1.put()
+    dispose_of(entity1.key._key)
+
+    entity2 = Dummy(foo="child", parent=parent_key, namespace=None)
+    entity2.put()
+    dispose_of(entity2.key._key)
+
+    entity3 = Dummy(foo="childless", namespace="xyz")
+    entity3.put()
+    dispose_of(entity3.key._key)
+
+    with client_context.new(namespace=other_namespace).use():
+        query = Dummy.query(ancestor=parent_key, namespace="xyz")
+        results = eventually(query.fetch, length_equals(2))
+
+    assert results[0].foo == "bar"
+    assert results[1].foo == "child"
+
+
+def test_ancestor_query_with_default_namespace(
+    client_context, dispose_of, other_namespace
+):
+    class Dummy(ndb.Model):
+        foo = ndb.StringProperty(default="")
+
+    entity1 = Dummy(foo="bar", namespace="")
+    parent_key = entity1.put()
+    dispose_of(entity1.key._key)
+
+    entity2 = Dummy(foo="child", parent=parent_key, namespace=None)
```

Comment:
I noticed in the customer's example that they don't explicitly pass `namespace`. I would think that the namespace would be implied by passing in `parent`. For grins, I tried removing the `namespace` arg from this test and I get a namespace mismatch exception in the model constructor. Is there maybe a little more work to do here?

Reply:
Well, I confess I added this to make the test pass, but the customer's code does run unchanged if using a python script outside of the test machinery. Maybe I'm missing something on the test setup?

Reply:
It looks like it's just because in the customer's example the context's namespace is still the default namespace when that entity is created, so there won't be a mismatch between entity2 and its parent. Does still seem like specifying parent should imply the namespace, though.

I know you're starting your new job. Would you like me to take this one?

Reply:
Yes, please take this. I appreciate the offer.

FWIW, the following code caused the mismatch error before the fix was applied, and it works fine with the fix in:

```
import os
from google.cloud import ndb

class Dummy(ndb.Model):
    foo = ndb.StringProperty(default='')

with ndb.Client().context():
    entity1 = Dummy(foo='bar', namespace='')
    entity1.put()
    key_entity1 = entity1.key

    entity2 = Dummy(foo='child', parent=key_entity1)
    entity2.put()

    entity3 = Dummy(foo='childless')
    entity3.put()

with ndb.Client().context(namespace='other_context'):
    l = Dummy.query(ancestor=key_entity1, namespace='').fetch()
    for e in l:
        print(e)

```

I couldn't quite duplicate that in the text.